lib/exercises.rb

```diff
@@ -1,20 +1,82 @@
-
+require 'awesome_print'
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
-
+# Time Complexity: O(nm) because the method will always iterate through each string (n) in strings and then each string will be compared to all hashes (m) that have been created, which in the worst case could be the same as the number of strings
+# Space Complexity: O(n) because the size of anagrams_grouped and hash_groups will grow linearly with the size of strings in the worse case
 def grouped_anagrams(strings)
```

Comment:
This works with good time/space complexity, but could use with some refactoring.